feat: add RW verbs for login rules to preset editor role

nklaassen opened the pull request.

nklaassen committed:
feat: add RW verbs for login rules to preset editor role (#20648)

kimlisa reviewed (approved):
out of curiosity what's a LoginRule?

nklaassen commented:
> out of curiosity what's a LoginRule?

@kimlisa https://github.com/gravitational/teleport/blob/master/rfd/0078-login-rules.md

nklaassen committed:
feat: add login rule audit event types (#20649)

This commit adds the types necessary in order to add audit events for
login rule resource actions.
The emission of these events will be added in a separate commit to
teleport.e, and the frontend display will be added later (currently
these show up as "Unknown" events).

nklaassen commented:
I accidentally merged https://github.com/gravitational/teleport/pull/20649 into this branch, but both changes were approved and are relatively small so I'm going to merge them to master together in this PR rather than try to separate them again
